service/history/ndc/workflow_resetter.go

```diff
@@ -873,3 +876,18 @@ func IsTerminatedByResetter(event *historypb.HistoryEvent) bool {
 	}
 	return false
 }
+
+// shouldExcludeAllReapplyEvents returns true if the excludeTypes map contains all the elegible re-apply event types.
+func (r *workflowResetterImpl) shouldExcludeAllReapplyEvents(excludeTypes map[enumspb.ResetReapplyExcludeType]bool) bool {
+	allExcludableEvents := []enumspb.ResetReapplyExcludeType{
+		enumspb.RESET_REAPPLY_EXCLUDE_TYPE_SIGNAL,
```

Comment:
Should we have an ALL enum value so we don't have to check every individual type? How will we remember to update this function every time we add another reapply type?
Can we at least build the slice in a static var based on `maps.Keys(enums.ResetReapplyExcludeType_name)` and exclude `enumspb.RESET_REAPPLY_EXCLUDE_TYPE_UNSPECIFIED`?

Reply:
+1 or maybe write a test that will fail if this list is not updated when new value is defined.

I think originally we decided to go with the ALL enum value, but looks like we are no longer doing that?

Reply:
Yes. That was the [original change here](https://github.com/temporalio/api/pull/454#issuecomment-2377918620). But @alexshtin wanted to provide explicit exclude option for each event type instead of a general exclude all. The reasoning was that it's somewhat confusing if someone provided `EXCLUDE_ALL` + `EXCLUDE_SIGNAL` options for example. Plus there is a deprecated option that lets user specify an event to be explicitly included. So we have a [function to merge options](https://github.com/temporalio/temporal/blob/main/service/history/api/resetworkflow/api.go#L173-L199) which gets complicated if we add `EXCLUDE_ALL`.

Reply:
> Can we at least build the slice in a static var based on maps.Keys(enums.ResetReapplyExcludeType_name) and exclude enumspb.RESET_REAPPLY_EXCLUDE_TYPE_UNSPECIFIED?
@bergundy that's a good idea. I'll update it if we decide to stick with not providing a EXCLUDE_ALL option.

Reply:
I feel if the new option is specified then we should just ignore the old one. EXCLUDE_ALL + EXCLUDE_SIGNAL, maybe just me, but I don't feel it's that confusing TBH.

I am ok with the new Nexus option, as long as we have a way to skip reapply entirely. Though we'll have to keep updating that enum for any new reapply types.

> allow them to specify a list of event types or super types (ex: NEXUS_EVENTS) to include during reset

hmm, the reason it's an exclusion is to provide backward compatibility, since by default everything will be reapplied.